src/node/util.ts

```diff
@@ -115,10 +118,162 @@ export const generatePassword = async (length = 24): Promise<string> => {
   return buffer.toString("hex").substring(0, length)
 }
 
-export const hash = (str: string): string => {
+/**
+ * Used to hash the password.
+ */
+export const hash = async (password: string): Promise<string> => {
+  try {
+    return await argon2.hash(password)
+  } catch (error) {
+    logger.error(error)
+    return ""
+  }
+}
+
+/**
+ * Used to verify if the password matches the hash
+ */
+export const isHashMatch = async (password: string, hash: string) => {
+  try {
+    return await argon2.verify(hash, password)
+  } catch (error) {
+    logger.error(error)
+    return false
+  }
+}
+
+/**
+ * Used to hash the password using the sha256
+ * algorithm. We only use this to for checking
+ * the hashed-password set in the config.
+ *
+ * Kept for legacy reasons.
+ */
+export const hashLegacy = (str: string): string => {
   return crypto.createHash("sha256").update(str).digest("hex")
 }
 
+/**
+ * Used to check if the password matches the hash using
+ * the hashLegacy function
+ */
+export const isHashLegacyMatch = (password: string, hashPassword: string) => {
+  const hashedWithLegacy = hashLegacy(password)
+  return safeCompare(hashedWithLegacy, hashPassword)
+}
+
+const passwordMethods = ["SHA256", "ARGON2", "PLAIN_TEXT"] as const
+export type PasswordMethod = typeof passwordMethods[number]
+
+/**
+ * Used to determine the password method.
+ *
+ * There are three options for the return value:
+ * 1. "SHA256" -> the legacy hashing algorithm
+ * 2. "ARGON2" -> the newest hashing algorithm
+ * 3. "PLAIN_TEXT" -> regular ol' password with no hashing
+ *
+ * @returns {PasswordMethod} "SHA256" | "ARGON2" | "PLAIN_TEXT"
+ */
+export function getPasswordMethod(hashedPassword: string | undefined): PasswordMethod {
```

Comment:
I'm not sure where in the code to put it, but it would be nice to issue a warning if someone is using the plain text/cookie/sha256 methods to tell them that their authentication method is insecure, and to consider upgrading to argon2.

Feel free to do that as a separate PR, though!

Reply:
Oh man, that is a fantastic idea! Then it would inform the ones using the sha256 that a better method exists! I'll do this in a follow-up PR 🙌 Thanks for the suggestion!

Reply:
https://github.com/cdr/code-server/issues/3560